data/test/scenarios/test_lua_name.cfg

```diff
@@ -0,0 +1,45 @@
+
+{GENERIC_UNIT_TEST test_lua_event_name (
+    [event]
+        name=start
+        [fire_event]
+            name=test
+        [/fire_event]
+    [/event]
+    [event]
+        name=test
+        action=<<
+            local actions = ...
+            local ret, err = pcall(function()
+                local some_variable = bar
+            end)
+            unit_test.assert_contains(err, 'test', 'error contains test')
+            unit_test.succeed()
+        >>
+    [/event]
+)}
+
+{GENERIC_UNIT_TEST test_lua_name (
```

Comment:
Tests should have documentation about what they're testing, otherwise it makes it hard to review whether the test is correct. This test is going to be broken by later changes in this PR, but let's get to that later.

Having asked again for advantages of `[event]action=` over `[event][lua]code=`, I got the response in IRC
> It does allow you to make a "safe" event that catches and handles errors, seems possibly helpful in unit tests at least.

which implied that this feature could only be done with `[event]action=`. It can also be done with `[event][lua]code=`, noting that the already-present `[lua]` tag needs to go inside the new `[lua]` tag's `[args]`.

On to why this test is going to break before this PR is complete: the `[lua]` tag that's already present looks like ActionWML. While he's been more open to `[event]action=` that I, @Pentarctagon has said that it needs to generate a warning in that case.

Reply:
The use in this test is not one that generates a warning. A warning is generated if both of the following conditions are satisfied:

* The Lua code does not contain the string `...`
* There are any tags in the event (excluding filter tags)

In this case, the Lua code _does_ contain `...`, so there's no warning. The warning is already implemented, and there is even a test that ensures it triggers. We could perhaps improve on this even further (it would be nice if `...` in a string or comment didn't count) but that might be a bit tricky.

> It can also be done with `[event][lua]code=`, noting that the already-present `[lua]` tag needs to go inside the new `[lua]` tag's `[args]`.

Ah yes, you're right. In fact, there's at least one test that already does this, possibly one of the for-loop tests? I don't quite remember which one it is, but it does exist.